samples/matter/common/src/diagnostic/diagnostic_logs_network.h

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#pragma once
+
+#include "diagnostic_logs_provider.h"
+#include "diagnostic_logs_retention.h"
+#include <lib/core/CHIPError.h>
+
+class DiagnosticLogsNetworkReader : public Nrf::Matter::DiagnosticLogsIntentIface {
```

Comment:
Hmm, I know that we talked about it but now I'm looking at this PR and it seems that both User Logs and Network Logs are exactly the same, but only have different constructor inputs. So why don't we resign with singleton here and allow the creation of multiple objects? It seems more proper in this case.

Reply:
I guess the reader also does the same things but on a different RAM space.

Reply:
Should the diagnostic logs provider contain those two objects?